infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Orchestrator.java

```diff
@@ -106,8 +135,11 @@ void addNode(CorfuPayloadMsg<OrchestratorRequest> msg, ChannelHandlerContext ctx
     @Nonnull
     private Workflow getWorkflow(@Nonnull OrchestratorRequest req) {
         Request payload = req.getRequest();
-        if (payload.getType().equals(Types.OrchestratorRequestType.ADD_NODE)) {
-            return new AddNodeWorkflow(payload);
+        switch (payload.getType()) {
+            case ADD_NODE:
```

Comment:
Can't you just make this part of the OrchestratorRequestType enum?

Add a new method:
OrchestratorRequestType.getWorkflow(OrchestratorRequest request)

Then no need to manage this extra switch statement.

Reply:
Done.